packages/web/src/common/store/cache/collections/utils/retrieveCollections.ts

```diff
@@ -91,18 +93,120 @@ export function* retrieveTracksForCollections(
   })
 }
 
+type retrieveCollectionArgs = {
+  playlistId?: Nullable<ID>
+  permalink?: Nullable<string>
+}
+
 /**
  * Retrieves a single collection via API client
  */
-export function* retrieveCollection(playlistId: ID) {
+export function* retrieveCollection({
+  playlistId,
+  permalink
+}: retrieveCollectionArgs) {
   yield* waitForRead()
   const apiClient = yield* getContext('apiClient')
-  const userId = yield* select(getUserId)
-  const playlists = yield* call([apiClient, 'getPlaylist'], {
-    playlistId,
-    currentUserId: userId
+  const currentUserId = yield* select(getUserId)
+  if (playlistId) {
+    const playlists = yield* call([apiClient, 'getPlaylist'], {
+      currentUserId,
+      playlistId
+    })
+    return playlists
+  }
+  if (permalink) {
+    const playlists = yield* call([apiClient, 'getPlaylistByPermalink'], {
+      currentUserId,
+      permalink
+    })
+    return playlists
+  }
+}
+
+function* getEntriesTimestamp(ids: (ID | string)[]) {
+  const selector = (state: CommonState, ids: (ID | string)[]) =>
+    ids.reduce(
+      (acc: { [id: number | string]: number | null }, id: ID | string) => {
+        acc[id] = getEntryTimestamp(state, { kind: Kind.COLLECTIONS, id })
+        return acc
+      },
+      {}
+    )
+  const selected = yield* select(selector, ids)
+  return selected
+}
+
+export function* retrieveCollectionByPermalink( // optional owner of collections to fetch (TODO: to be removed)
+  permalink: string,
+  // whether or not to fetch the tracks inside the playlist
+  fetchTracks = false,
+  /**  whether or not fetching this collection requires it to have all its tracks.
+   * In the case where a collection is already cached with partial tracks, use this flag to refetch from source.
+   */
+  requiresAllTracks = false
+) {
+  // @ts-ignore retrieve should be refactored to ts first
+  const { entries, uids } = yield* call(retrieve, {
+    ids: [permalink],
+    selectFromCache: function* (permalinks: string[]) {
+      const res = yield* select(cacheCollectionsSelectors.getCollections, {
+        permalinks
+      })
+      if (requiresAllTracks) {
+        const keys = Object.keys(res) as any
```

Comment:
so probably best to type in more strictly, to `as number[]`. this is tricky though cause i think object.keys always returns strings

````
const a = {1: 1, 2: 2}
Object.keys(a)
==> ['1', '2']
```
i think if you just let object.keys be string[] and remove `collectionId: number` to be `collectionId` untyped, should still work out same.